pkgs/stdenv/generic/check-meta.nix

```diff
@@ -115,6 +111,97 @@ let
   showLicense = showLicenseOrSourceType;
   showSourceType = showLicenseOrSourceType;
 
+  # Problem handler levels
+  problemHandlers = [ "error" "warn" "ignore" ];
+  # Take the maximum (highest severity) problem handler
+  maxProblemHandler =
+    a: b:
+    if a == "error" || b == "error" then
+      "error"
+    else if a == "warn" || b == "warn" then
+      "warn"
+    else
+      "ignore"
+  ;
+
+  # Currently known list of known problem kinds. Keep up to date with pkgs/top-level/config.nix
+  problemKinds = [
+    "removal"
+    "deprecated"
+    "maintainerless"
+    "insecure"
+    "broken"
+    "unsupported"
+  ];
+  # Problem kinds that are currently allowed in `meta.problems`
+  problemKindsManual = [
+    "removal"
+    "deprecated"
+  ];
+  # Problem kinds that are only allowed up to once per package
+  problemKindsUnique = [
+    "removal"
+    "maintainerless"
+  ];
+  # Same thing but a set with null values (comes in handy at times)
+  problemKindsUnique' = lib.pipe problemKindsUnique [
+    (map (name: { inherit name; value = null; }))
+    lib.listToAttrs
+  ];
+
+  # Extract a list of problems from the derivation. Every problem has at least a `kind` and `message` field.
+  # generateProblems :: Derivation -> [{ name, kind, message, handler, … }]
+  generateProblems =
+    pkg:
+    let
+      pkgName = lib.getName pkg;
+
+      problems =
+        (pkg.meta.problems or {})
+        // lib.optionalAttrs (hasNoMaintainers pkg) {
+          maintainerless = {
+            message = "This package has an empty `meta.maintainers`";
+          };
+        }
+      ;
+
+      # Inject default values, since metaTypes can't do it for us currently
+      addDefaults = name: problem: { inherit name; kind = name; } // problem;
+
+      # Determine the handler level from config for this problem
+      addHandler = problem:
+        let
+          handler =
+            # Try to find an explicit handler
+            (config.problems.handlers.${pkgName} or {}).${problem.name}
+            # Fall back, iterating through the matchers
+            or (
+              lib.pipe (/*builtins.deepSeq config.problems.matchers*/ config.problems.matchers) [
```

Comment:
cleanup the previous attempts if they are not relevant anymore? or add a comment